src/log_surgeon/finite_automata/RegexDFAStatePair.hpp

```diff
@@ -65,15 +65,15 @@ class RegexDFAStatePair {
 
 template <typename DFAState>
 auto RegexDFAStatePair<DFAState>::get_reachable_pairs(
-        std::set<RegexDFAStatePair<DFAState>>& visited_pairs,
-        std::set<RegexDFAStatePair<DFAState>>& unvisited_pairs
+        std::set<RegexDFAStatePair>& visited_pairs,
+        std::set<RegexDFAStatePair>& unvisited_pairs
 ) const -> void {
     // TODO: Handle UTF-8 (multi-byte transitions) as well
     for (uint32_t i = 0; i < cSizeOfByte; i++) {
         auto next_state1 = m_state1->next(i);
         auto next_state2 = m_state2->next(i);
         if (next_state1 != nullptr && next_state2 != nullptr) {
-            RegexDFAStatePair<DFAState> reachable_pair{next_state1, next_state2};
+            RegexDFAStatePair reachable_pair{next_state1, next_state2};
```

Comment:
Nit:
```suggestion
            RegexDFAStatePair const reachable_pair{next_state1, next_state2};
```
I understand this is out of the PR's scope, but I don't mind if we also include fixes like this
It's up to u whether to commit this change within this PR